### ISSUE TITLE
Improve Documentation of masker family

### ISSUE THREAD

**Issue by banilo** [nilearn/nilearn#355]
Perhaps the docstrings for the following classes could be improved:

```
- NiftiMasker
- NiftiLabelsMasker
- NiftiMapsMasker
- MultiNiftiMasker
```

More specifically, at the top of the respective class docstrings, one might add
- a more generous explanation of the purpose of the class
- its commonalities and differences comparing to the respective other masker classes
- provide one simple use case when this is typically useful

Apart from that, a new nilearn example might be useful that uses the various masker and, thus, gives an overview of this part of the API.

Edited by @AlexandreAbraham:

Here is the list of remarks made on _Nifti_Masker doc across issues:
- [ ] Specify which data is detrended (#418)
- [ ] Add intersect_masks to NiftiMasker's "related functions" (#436)
- [ ] Specify that only `mask_args` and `mask_strategy` are used for fitting. It's specified in the class docstring, but not on the fit function docstring (#436). Gael suggested to add it in the Notes section.
- [ ] Add in transform (#436):
  - Parameters (detrend, standardize) are computed from the images passed to transform (i.e. not computed and stored from fit)
  - Whether the inverse of these parameters is applied when inverse_transform is called (not sure if they are, for detrend, standardize, and confounds).
- [ ] NiftiMapsMasker and NiftiLabelsMasker don't have docstring for fit_transforms (#603).

**Comment by lesteve** [comment]
There is something like this in the online doc: http://nilearn.github.io/building_blocks/data_preparation.html.

Maybe it could be expanded upon?

**Comment by GaelVaroquaux** [comment]
Actually, it's less about expanding and more about making it easy to find. Right now, a problem of the documentation is that it lack a big picture organization. I think that we should rework that during the next sprint. How about beginning of February?

**Comment by AlexandreAbraham** [comment]
> How about beginning of February?
:+1: for a cosmit sprint to make the whole package homogeneous and well documented before the release.

**Comment by AlexandreAbraham** [comment]
According to #418, we should specify exactly which data is detrended.

**Comment by AlexandreAbraham** [comment]
@bcipolli's comments from #436 

1. Please add `intersect_masks` to `NiftiMasker`'s "related functions" would be appreciated.

2. I suggest adding to the docstring of fit, to reminds users that only `mask_args` and `mask_strategy` are used for fitting.  It's specified in the class docstring, but not on the `fit` function docstring.

3. Lastly, I would find it helpful to specify (here or in transform) that:
    * Parameters (detrend, standardize) are computed from the images passed to transform (i.e. not computed and stored from fit)
    * Whether the inverse of these parameters is applied when inverse_transform is called (not sure if they are, for detrend, standardize, and confounds).

    My reasoning for this: my lab does computer vision, and the setup here differs from how our lab (and I believe others in this area) preprocess our datasets.  We compute and store whitening transforms for our training set, then apply them to test and validation sets (which are assumed to come from the same distribution of images as the training set).  I think very clear documentation about the use of these maskers could help avoid confusion for people like me coming from other fields as to the intended use of these maskers.

**Comment by GaelVaroquaux** [comment]
>  2. I suggest adding to the docstring of fit, to reminds users that only
>     mask_args and mask_strategy are used for fitting. It's specified in the
>     class docstring, but not on the fit function docstring.

In the 'notes' section.

>     I think very clear documentation about the use of these maskers
>     could help avoid confusion for people like me coming from other
>     fields as to the intended use of these maskers.

I wonder if this should not be in the narrative documentation of the
maskers, rather than in the docstring. One of the reasons is that I am
afraid of duplication of the information.

**Comment by banilo** [comment]
We still do not have an example that does in-your-face-teach the commonalities and differences of the sociology of the nifti masker clan, or do we?

**Comment by GaelVaroquaux** [comment]
> We still do not have an example that does in-your-face-teach the commonalities
> and differences of the sociology of the nifti masker clan, or do we?

Nope. It might be good to have a notebook-like example on that.

**Comment by JohannesWiesner** [comment]
I would like to add a specific documentation issue concerning `NiftiMasker`. This is basically a copy of [my question](https://neurostars.org/t/how-does-niftimasker-work-when-dealing-with-mask-and-data-images-of-different-resolutions/4892) on neurostars.org. As a relatively new user, I wasn't sure how `NiftiMasker` works when dealing with mask and data images of different shapes/affines/resolutions and how this interplays with `target_affine` or `target_shape`. I find the `NiftiMasker` class a little bit underdocumented. @jeromedockes gave a good explanation and also mentioned that this should be stated more explicitly in the documentation. Maybe this could be added to the documentation of the `NiftiMasker` class?

**Comment by bthirion** [comment]
Would you contribute it ? That would be awesome ...
Thx

**Comment by bthirion** [comment]
Stalled. Closing.
